---
title: "Pu'u Eke - Microbial Figures"
author: "Emily Lacroix"
date: "9/23/2021"
output: 
  github_document: 
    toc: TRUE
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Set-up

## libraries
```{r}
library(tidyverse)
library(readxl)
library(phyloseq)
library(ggrepel)
```


## files
```{r eval=FALSE}
phyloseq_object_file <- "MPC_16S_phyloseq_2019_07_23.rds"

parent_data_file <- "GCB2021_PuuEke_AllData.xlsx"
```


```{r include=FALSE}
phyloseq_object_file <- "~/Box Sync/Stanford (elacroix@stanford.edu)/Research/AnoxicMicrositesHawaii/GCBSubmission/Data/MPC_16S_phyloseq_2019_07_23.rds"

parent_data_file <- "~/Box Sync/Stanford (elacroix@stanford.edu)/Research/AnoxicMicrositesHawaii/GCBSubmission/Data/GCB2021_PuuEke_AllData.xlsx"

```

# Analysis

## Data clean-up

### Filter and subset
Filter and subset phyloseq object to exclude rare taxa (< 200) and create subsets that focus solely on the upper horizons for asv analysis
```{r}
#read in object
raw_phyloseq_object <- phyloseq_object_file %>% read_rds()

#filter out low frequency (<200) reads
phyloseq_trimmed <- 
  subset_samples(raw_phyloseq_object, sample_sums(raw_phyloseq_object) > 200)

#create a object that encompasses just the upper horizons
phyloseq_upperhorizons <-   
  subset_samples(phyloseq_trimmed,horizon %in% c("1_Bh1", "2_Bhg", "3_Bh2")) 

# Convert asv to report relative abundance of the upper horizons
phyloseq_upperhzn_rel_abund <- 
  transform_sample_counts(phyloseq_upperhorizons, function(x) x / sum(x))

# Turn phyloseq object into a flattened tibble for easy plotting 

flattened_phyloseq_obj <- #rename to flattened ps obj
  psmelt(phyloseq_upperhzn_rel_abund) %>% 
  select(
    asv = OTU, 
    sample = Sample, 
    rel_abund = Abundance
  ) %>% 
  pivot_wider(id_cols = asv, names_from = sample, values_from = rel_abund)

```


Save off the taxonomic information 
```{r}
taxonomic_table <- 
  phyloseq_upperhzn_rel_abund %>% 
  tax_table() %>% 
  as("matrix") %>% 
  as_tibble(rownames = "OTU") %>% 
  rename(asv = OTU)


```

### Sample Info
Create sample information table from phyloseq object; create one that includes counts from 4 horizons and another that contains sample metadata for the upper horizons

```{r warning=FALSE}
all_sample_data <- 
  sample_data(raw_phyloseq_object) %>% 
  as_tibble(rownames = "sample")

sample_data <- 
  sample_data(phyloseq_upperhzn_rel_abund) %>% 
  as_tibble(rownames = "sample")


```

### Putative Anaerobes
Read-in the ASVs identified as putative anaerobes per RAxML tree
```{r warning}
putative_anaerobes <- 
  parent_data_file %>% 
  read_xlsx(sheet = "PutativeAnaerobes") %>% 
  select(asv = ASV) %>% 
  mutate(anaerobe = TRUE)

```

# Figures
## Figure 5a: 16S rRNA Gene Copies

```{r}

all_sample_data %>% 
  filter(!is.na(rRNA_16S_copy_number_per_g_soil_logtrans)) %>% 
  ggplot(
    aes(
      x = distance_mm, 
      y = rRNA_16S_copy_number_per_g_soil_logtrans, 
      shape = horizon
    )
  ) + 
  geom_line(linetype = 2, alpha = 0.6, size = 0.5) +
  geom_point(size = 3) + 
  scale_x_continuous(
    breaks = seq(from = 0, to = 10, by = 2),
    limits = c(-0.1, 10),
    expand = expansion(mult = c(0, 0.05))
  ) + 
  scale_y_continuous(
    breaks = seq(from = 4, to = 14, by = 1),
    limits = c(4, 11)
  ) + 
  scale_shape_manual(
    values = c(16, 2, 15, 1, 17)
  ) + 
  theme_bw() + 
  theme(
    strip.background = element_blank(),
    strip.text.x = element_blank(),
    panel.grid = element_blank(),
    aspect.ratio = 1.2,
    legend.position = "none",
    axis.text = element_text(size = 14),
    axis.title = element_text(size = 16)
  ) + 
  labs(
    x = "Distance from preferential flow paths (mm)",
    y = "Log Copy Number/ g soil"
  ) 

  

```


## Figure 5b: Relative Abundance of putative anaerobes
```{r}
flattened_phyloseq_obj %>% 
  left_join(putative_anaerobes, by = "asv") %>% 
  mutate_at(
    vars(anaerobe),
    ~ if_else(is.na(.), FALSE, TRUE)
  ) %>% 
  filter(anaerobe == TRUE) %>% 
  select(-asv) %>% 
  pivot_longer(
    cols = contains("fastq"), 
    names_to = "sample", 
    values_to = "relative_abundance"
  ) %>% 
  group_by(sample) %>% 
  summarise(relative_anaerobe = sum(relative_abundance)) %>% 
  left_join(sample_data, by = "sample") %>% 
  mutate_at(vars(distance_mm), as.numeric) %>% 
  mutate_at(
    vars(horizon), 
    ~ factor(
      .,
      levels = 
        c(
          "1_Bh1", 
          "2_Bhg", 
          "3_Bh2"
        ),
      labels = 
        c(
          "Bh1", 
          "Bhg", 
          "Bh2"
        )
    )
  ) %>% 
  ggplot(aes(x = distance_mm, y = relative_anaerobe, shape = horizon)) + 
  geom_line(linetype = 2, alpha = 0.5) + 
  geom_point(size = 3) + 
  scale_y_continuous(
    breaks = c(0.02, 0.04, 0.06, 0.08, 0.10, 0.12),
    labels = scales::percent_format(accuracy = 1)
  ) + 
  scale_x_continuous(
    breaks = seq(from = 0, to = 10, by = 2),
    limits = c(-0.1, 10),
    expand = expansion(mult = c(0, 0.05))
  ) +
  scale_shape_manual(
    values = c(16, 2, 15, 1, 17)
  ) + 
  theme_bw() + 
  theme(
    strip.background = element_blank(),
    strip.text.x = element_blank(),
    panel.grid = element_blank(),
    aspect.ratio = 1.2,
    legend.position = "none",
    axis.text = element_text(size = 14),
    axis.title = element_text(size = 16)
  ) + 
  labs(
    x = "Distance from preferential flow paths (mm)",
    y = "% Abundance",
    shape = NULL
  ) 
```

### ID the 2 dominant putative anaerobe phyla
```{r}  

flattened_phyloseq_obj %>% 
  left_join(putative_anaerobes, by = "asv") %>% 
  mutate_at(
    vars(anaerobe),
    ~ if_else(is.na(.), FALSE, TRUE)
  ) %>% 
  filter(anaerobe == TRUE) %>% 
  pivot_longer(
    cols = contains("fastq"), 
    names_to = "sample", 
    values_to = "relative_abundance"
  ) %>% 
  group_by(sample) %>% 
  slice_max(relative_abundance, n = 2) %>% 
  left_join(taxonomic_table, by = "asv") %>% 
  left_join(sample_data, by = "sample") %>% 
  mutate_at(vars(distance_mm), as.numeric) %>% 
  arrange(horizon, distance_mm) %>% 
  select(horizon, distance_mm, Kingdom, Phylum, relative_abundance, asv)
  
```



## Figure SI5: NMDS
```{r}
set.seed(123)

my.ord <- ordinate(phyloseq_upperhorizons, "NMDS", "bray")

plot_ordination(
  phyloseq_upperhorizons, 
  my.ord, 
  shape = "horizon"
) + 
  geom_point(size = 3) +
  geom_text_repel(
    mapping = aes(label = str_c(distance_mm, " mm")),
  ) +
  scale_shape_manual(
    values = c(16, 2, 15)
  ) + 
  scale_x_continuous(expand = c(0.1,0.1)) + 
  theme_bw() + 
  theme(
    strip.background = element_blank(),
    strip.text.x = element_blank(),
    panel.grid = element_blank(),
    aspect.ratio = 1,
    legend.position = "none",
    axis.text = element_text(size = 14),
    axis.title = element_text(size = 16)
  ) + 
  labs(
    shape = NULL
  ) 

```

## Figure SI6: Abundant phyla

These are the 10 most abundant phyla in each sample

```{r}
top_10_phyla <- 
  flattened_phyloseq_obj %>% 
  left_join(taxonomic_table, by = "asv") %>% 
  pivot_longer(
    cols = contains("fastq"), 
    names_to = "sample", 
    values_to = "relative_abundance"
  ) %>%
  group_by(sample, Kingdom, Phylum) %>% 
  summarise(relative_phylum_abundance = sum(relative_abundance)) %>%
  left_join(sample_data, by = "sample") %>% 
  group_by(sample) %>% 
  slice_max(relative_phylum_abundance, n = 10) %>% 
  mutate_at(vars(distance_mm), as.factor) %>% 
  mutate_at(
    vars(Phylum),
    ~ case_when(
      is.na(.) & Kingdom == "Bacteria" ~ "Unidentified Bacteria",
      is.na(.) & Kingdom == "Archaea" ~ "Unidentified Archaea",
      TRUE ~ .
      )
  ) 
  
```



```{r}
top_10_phyla %>% 
  ggplot(aes(x = distance_mm, y = relative_phylum_abundance, fill = Phylum)) + 
  geom_col() + 
  scale_fill_viridis_d() +
  scale_y_continuous(breaks = seq(0, 1, 0.2)) + 
  facet_grid(rows = vars(horizon)) + 
  theme_bw() + 
  labs(
    x = "Distance from preferential flow path (mm)",
    y = "Relative Abundance"
  ) 
```







