---
title: "Compositing Data"
author: "Emily Lacroix"
date: "1/21/2021"
output: github_document
---

This script is meant to composite all of the different data sources given to me (Emily) from Scott, Glade, and Yoko. This data composite will be posted to github upon publication. This script is for personal use - not for distribution. 

# Set-up

## knitr

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## libraries

```{r}
library(tidyverse)
library(readxl)
```

## files

```{r}
data_file_2011 <- "~/Box Sync/Stanford (elacroix@stanford.edu)/Research/AnoxicMicrositesHawaii/GCBSubmission/Data/SOM_PFP_2011 summary_eml.xlsx"

data_file_2012 <- "~/Box Sync/Stanford (elacroix@stanford.edu)/Research/AnoxicMicrositesHawaii/GCBSubmission/Data/SOM_PFP_2012_summary_eml.xlsx"

taxonomy_file <- "~/Box Sync/Stanford (elacroix@stanford.edu)/Research/AnoxicMicrositesHawaii/GCBSubmission/Data/MPC_Functional_Analysis_2019_07_26_eml.xlsx"

raxml_file <- "~/Box Sync/Stanford (elacroix@stanford.edu)/Research/AnoxicMicrositesHawaii/GCBSubmission/Data/Scott_Manuscript_Tree/Archaea_List_Euryarchaeota_Clade_2019_09_12.xlsx"

qpcr_file <- "~/Box Sync/Stanford (elacroix@stanford.edu)/Research/AnoxicMicrositesHawaii/GCBSubmission/Data/qPCR data to Scott and Oliver/qPCR_summary_final.xlsx"

cnexafs_file <- "~/Box Sync/Stanford (elacroix@stanford.edu)/Research/AnoxicMicrositesHawaii/GCBSubmission/Data/CNEXAFS.xlsx"
```


## File output
```{r}
output_csv <- "~/Box Sync/Stanford (elacroix@stanford.edu)/Research/AnoxicMicrositesHawaii/GCBSubmission/Data/parent_data_table.csv"
```


# Import data
## Read-in 2011 Data
```{r}
data_2011 <- 
  read_xlsx(data_file_2011, sheet = "Sheet1") %>% 
  filter(!is.na(`sample ID`)) %>% 
  select(
    horizon = Horizon,
    depth = Depth,
    sample_id = `sample ID`,
    pfp_distance_mm = `Distance from PFP (mm)`,
    avg_dist_mm = `average distance (mm)`,
    perc_c = `C%`,
    perc_n = `N%`,
    c_to_n = `C:N`,
    c14 = `∆14C ‰`,
    c14_age = `14C age (BP)`,
    sro_minerals_mmol_kg = `ammonium ox (Al+Fe+Si) mol/Kg`
  ) %>% 
  mutate(
    collection_yr = 2011
  )

```

## Read-in 2012 Data
```{r}
data_2012 <- 
  read_xlsx(data_file_2012, sheet = "Sheet2") %>% 
  filter(!is.na(`sample ID`)) %>% 
  select(
    horizon = Horizon,
    depth = Depth,
    sample_id = `sample ID`,
    pfp_distance_mm = `Distance from PFP (mm)`,
    avg_dist_mm = `average distance (mm)`,
    perc_c = `C%`,
    perc_n = `N%`,
    c_to_n = `C:N`,
    c14 = `∆14C ‰`,
    c14_age = `14C age (BP)`,
    sro_minerals_mmol_kg = `ammonium ox (Fe+Al+Si) mmol/kg soils`
  ) %>% 
  mutate(collection_yr = 2012)

```


## Read in microbial data

### 16S rRNA abundance data
```{r}
qpcr_data <- 
  qpcr_file %>% 
  read_xlsx(sheet = "Sheet1", skip = 1) %>% 
  select(
    sample_id = `sample ID`,
    pfp_distance_mm = `Distance from PFP (mm)`, 
    copy_number_16s_rrna_per_kg = `copy number / soil (Kg)...21`
  ) %>% 
  filter(!is.na(sample_id)) %>% 
  #mutate_at(vars(sample_id), ~ as.integer(str_remove_all(.,"-"))) %>% 
  mutate(collection_yr = 2012)


```



### Calculate relative abundance of putative anaerobes for each sample
```{r}
rdp_taxonomy_table <-
  taxonomy_file %>% 
  read_xlsx(sheet = "RDP Taxonomy Table")
```

```{r}
putative_anaerobes <- 
  raxml_file %>% 
  read_xlsx(sheet = "PutativeAnaerobes") %>% 
  select(ASV) %>% 
  mutate(anaerobe = TRUE)

```

```{r}

sample_identity <-
  taxonomy_file %>% 
  read_xlsx(sheet = "Summary", range = "B2:AA4") %>% 
  pivot_longer(
    cols = contains("fastq"),
    names_to = "sample",
    values_to = "value"
  ) %>% 
  pivot_wider(
    id_cols = "sample",
    names_from = "Sample_Name",
    values_from = "value"
  ) %>% 
  rename_all(str_to_lower) %>% 
  mutate(
    sample_id = as.integer(
      str_extract(str_extract(sample, ".\\d{2}.R1.fastq"), "\\d{2}")
      )
  )
```

```{r}
relative_anaerobe <- 
  rdp_taxonomy_table %>% 
  left_join(putative_anaerobes, by = "ASV") %>% 
  mutate_at(
    vars(anaerobe),
    ~ if_else(is.na(.), FALSE, TRUE)
  ) %>% 
  filter(anaerobe == TRUE) %>% 
  select(-ASV) %>% 
  pivot_longer(
    cols = contains("fastq"), 
    names_to = "sample", 
    values_to = "relative_abundance"
  ) %>% 
  group_by(sample) %>% 
  summarise(relative_anaerobe = sum(relative_abundance)) %>% 
  left_join(sample_identity, by = "sample") %>% 
  mutate_at(vars(distance_mm), as.numeric) %>% 
  filter(horizon %in% c("Bh1", "Bhg", "Bh2")) %>%
  mutate_at(
    vars(horizon), 
    ~ factor(
      .,
      levels = 
        c(
          "Bh1", 
          "Bhg", 
          "Bh2"
        )
    )
  ) %>% 
  mutate(collection_yr = 2012) %>% 
  select(
    collection_yr,
    horizon,
    distance_mm,
    relative_anaerobe
  )

```

## C-NEXAFS Table
```{r}
cnexafs_data <- 
  cnexafs_file %>% 
  read_xlsx(sheet = "Sheet1")
```


# Combine data sets into one
```{r}

data <-
  data_2011 %>% 
  bind_rows(data_2012) %>% 
  left_join(
    qpcr_data, 
    by = c("sample_id", "collection_yr", "pfp_distance_mm")
  ) %>% 

  select(
    collection_yr, 
    horizon,
    depth, 
    pfp_distance_mm, 
    distance_mm = avg_dist_mm, 
    perc_c, 
    perc_n,
    c_to_n, 
    c14, 
    sro_minerals_mmol_kg,
    copy_number_16s_rrna_per_kg
  ) %>% 
  left_join(
    relative_anaerobe,
    by = c("collection_yr", "horizon", "distance_mm")
  ) %>% 
    left_join(
    cnexafs_data, 
    by = 
      c(
        "collection_yr", 
        "horizon",
        "distance_mm"
      )
  ) 

data
```

# Write csv for Excel workbook
```{r}
#data %>% write_csv(file = output_csv)
```

